backend/fleet_management/models.py

```diff
@@ -19,15 +18,18 @@ class User(AbstractUser):
     rsa_pub_e = models.CharField(max_length=6, null=False, default="")
     rsa_priv_d = models.CharField(max_length=6, null=False, default="")
 
-    def save(self, regenerate_keys: bool = False, *args, **kwargs):
-        if regenerate_keys or self.pk is None:
-            pub, priv = find_pair_of_keys()
-            self.rsa_modulus_n = str(pub.n).zfill(6)
-            self.rsa_pub_e = str(pub.e).zfill(6)
-            self.rsa_priv_d = str(priv.d).zfill(6)
+    def save(self, *args, **kwargs):
+        if self.pk is None:
+            self.regenerate_keys()
```

Comment:
This means that `regenerate_keys` is called when user is created?

Reply:
Yes, it's called only when user is _created_, not _updated_. We don't want to regenerate keys after every change in their data (though technically we could regenerate it like that, it would still work but there's no point in doing this).